src/test/java/com/company/elixr/springbootapplicationwithdocker/FileOperationServiceTest.java

```diff
@@ -0,0 +1,164 @@
+        fileInfo.setId(UUID.randomUUID());
+        fileInfo.setFileName("testFile.txt");
+        fileInfo.setUserName("Nathan122");
+        fileInfo.setTimeOfUpload(LocalDateTime.now());
+
+        fileInfoList.add(fileInfo);
+    }
+
+    @Test
+    void test_SaveFile() {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.txt",
+                "text/plain",
+                "This is the file content".getBytes());
+        Mockito.when(fileOperationRepository.save(Mockito.any(FileInfo.class))).thenReturn(fileInfo);
+        ResponseEntity<SuccessResponse> created = fileOperationService.saveFile(sampleFile, "Natasha11");
+        Assertions.assertNotNull(created);
+        Assertions.assertEquals(HttpStatus.OK, created.getStatusCode());
+        Assertions.assertEquals(Constants.STATUS, created.getBody().getStatus());
+        Assertions.assertEquals(fileInfo.getId().getClass().getName(), created.getBody().getId().getClass().getName());
+    }
+
+    @Test
+    void test_SaveFile_InvalidFileType() {
+
+        MockMultipartFile sampleFile = new MockMultipartFile("file", "testFile.json",
+                "application/json",
+                "This is the file content".getBytes());
+        Mockito.when(fileOperationRepository.save(Mockito.any(FileInfo.class))).thenReturn(fileInfo);
+        try
+        {
+            fileOperationService.saveFile(sampleFile, "Natasha11");
+        } catch (BadRequestException exception) {
+            Assertions.assertNotNull(exception);
+            Assertions.assertEquals(Constants.ERROR_BAD_REQUEST_FILE_NOT_PRESENT_OR_INVALID_FILE_TYPE,
+                    exception.getMessage());
+        }
+    }
+
+    @Test
+    public void test_getFileById() {
+
+        UUID id = UUID.fromString("ebadf9b2-ea84-4683-bb32-a5bb4952e265");
+        Mockito.when(fileOperationRepository.findById(Mockito.any(UUID.class)))
+                .thenReturn(Optional.of(fileInfo));
+        ResponseEntity<SuccessResponseForGetById> found = fileOperationService
+                .findFileById(String.valueOf(id));
+        Assertions.assertNotNull(found);
+        Assertions.assertEquals(HttpStatus.OK, found.getStatusCode());
+        Assertions.assertEquals(2, found.getBody().getClass().getDeclaredFields().length);
+        Assertions.assertEquals(Constants.SUCCESS, found.getBody().getSuccess());
+        Assertions.assertEquals(fileInfo.getUserName(), found.getBody().getData().getUserName());
+        Assertions.assertEquals(fileInfo.getFileName(), found.getBody().getData().getFileName());
+    }
+
+    @Test
+    public void test_getFileById_IdNotInUUIDFormat() {
+
+        try {
+            String id = "ebadf9b2ea844683";
+        Mockito.when(fileOperationRepository.findById(Mockito.any(UUID.class)))
+                .thenThrow(new BadRequestException(Constants.ERROR_BAD_REQUEST_INVALID_ID_FORMAT));
+            fileOperationService.findFileById(id);
+        } catch (Exception exception) {
```

Comment:
Always catch the specific exception you are expecting

Reply:
Sure, will do.